src/documentation/chunk.rs

```diff
@@ -210,6 +222,23 @@ impl From<Clusters> for Vec<CheckableChunk> {
     }
 }
 
+/// Extract lines together with associated `Range`s relative to str `s`
+///
+/// Easily collectable into a `HashMap`.
+fn lines_with_ranges<'a>(s: &'a str) -> impl Iterator<Item = (Range, &'a str)> + Clone {
+    // @todo line consumes \r\n and \n so the ranges could off by 1 on windows
```

Comment:
Could we _just_ replace all `\r\n` with `\n` and vice versa?

Reply:
Not so simple, because the range to span mapping in the original file suffers an offset of 1

Reply:
But if we replace them right after opening the file before we start processing and just change it back before writing?

Reply:
Then we need to keep track of their spans plus the modifications we do, remember that they might be mixed insanely..

We really want to preserve everything, including spaces (which have markdown/cmark semantic meaning i.e. `  ` two spaces cause a soft break) whenever possible